docs/controllers/generics.md

```diff
@@ -0,0 +1,113 @@
+# Generics
+
+This chapter contains tips and tricks for re-using generic reconcilers.
+
+## Generic Reconcilers
+It is possible to create generic reconcilers by abstracing away the underlying type by using either [DynamicObject] or [PartialObjectMeta].
+
+This is a useful technique for controllers that need to do _the same thing_ to a bunch of resources, like for instance adding consistent labels / annotations.
+
+### PartialObjectMeta
+
+This is the easiest and cheapest way to create a generic reconciler because it still retains type information, and you do not have to devolve into a fully dynamic api.
+
+You can create generic reconcilers:
+
+```rust
+pub async fn reconcile<K>(obj: Arc<PartialObjectMeta<K>>, ctx: Arc<Context>)
+-> Result<Action>
+where
+    K: Resource<Scope = NamespaceResourceScope, DynamicType = ()>
+        + Clone
+        + DeserializeOwned
+        + Debug,
+{
+    let kind = K::kind(&()).to_string();
+    let ns = obj.namespace().unwrap();
+    let object_name = obj.name_any();
+    let api: Api<PartialObjectMeta<K>> = Api::namespaced(ctx.client, &ns);
+
+    // example work; apply some labels to the object
+    let patch: Patch<serde_json::Value> = get_standard_labels_for(&obj)?;
+    let serverside = PatchParams::apply("labeler");
+    api.patch(&object_name, &serverside, &patch).await?;
+
+    Ok(Action::requeue(Duration::from_secs(5 * 60)))
+}
+```
+
+The generic constraints on the associated type of the [Resource] here means this is a __namespaced__ resource (hence the unwrap). You could remove this bound (or change it for a cluster scoped only bound), but then you could not unwrap.
```

Comment:
didnt get what you mean by `generic constraints`.  tried to write in simpler words
```suggestion
The controller in the example runs for all resources of a specific `kind` in a particular namespace thus the scope __namespaced__. You could also remove the bound and run it for a cluster scope `Api::all` instead of `Api::namespaced`
```

Reply:
you have also made it wrong, ignoring this. 🙃